### PR TITLE
Add translated error message for running a managed task on an unmanaged thread

### PR DESCRIPTION
**Pull request by aguibert**
When running the `com.ibm.ws.concurrent.mp_fat` bucket locally, I was able to consistently reproduce the following error:

```
[11/14/18, 11:34:08:495 CST] 00000031 SystemErr                                                    R java.util.concurrent.ExecutionException: java.lang.SecurityException: setContextClassLoader
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022)
	at com.ibm.ws.concurrent.mp.ManagedCompletableFuture.get(ManagedCompletableFuture.java:771)
	at concurrent.mp.fat.web.MPConcurrentTestServlet.testAcceptEither(MPConcurrentTestServlet.java:307)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at componenttest.app.FATServlet.doGet(FATServlet.java:71)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:686)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1255)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:743)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:440)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1221)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1005)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:75)
	at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:83)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:927)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1023)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:417)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:376)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:532)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:466)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:331)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:302)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:165)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:74)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:501)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:571)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:926)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1015)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:232)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:825)
Caused by: java.lang.SecurityException: setContextClassLoader
	at java.base/java.util.concurrent.ForkJoinWorkerThread$InnocuousForkJoinWorkerThread.setContextClassLoader(ForkJoinWorkerThread.java:239)
	at com.ibm.ws.classloader.context.internal.ClassloaderContextImpl$2.run(ClassloaderContextImpl.java:154)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:647)
	at com.ibm.ws.classloader.context.internal.ClassloaderContextImpl.setCL(ClassloaderContextImpl.java:158)
	at com.ibm.ws.classloader.context.internal.ClassloaderContextImpl.taskStarting(ClassloaderContextImpl.java:133)
	at com.ibm.ws.context.service.serializable.ThreadContextDescriptorImpl.taskStarting(ThreadContextDescriptorImpl.java:387)
	at com.ibm.ws.concurrent.mp.ContextualConsumer.accept(ContextualConsumer.java:35)
	at java.base/java.util.concurrent.CompletableFuture$OrAccept.tryFire(CompletableFuture.java:1564)
	at java.base/java.util.concurrent.CompletableFuture$CoCompletion.tryFire(CompletableFuture.java:1144)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1705)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

At first I thought it was a Java 2 security issue, but then we realized it was caused Liberty trying to apply classloader context to an unprivileged unmanaged thread from the JVM that does not allow setting the TCCL.

This PR aims offer up a better error message for apps that run into this situation, because the message thrown by the JVM is very unhelpful.

### CODE REVIEW

File: dev/com.ibm.ws.classloader.context/resources/com/ibm/ws/classloader/context/internal/resources/ClassLoaderContextMessages.nlsprops [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#COMPONENTNAMEFOR CWWKL Class Loader Context
+#
+# NLS_MESSAGEFORMAT_VAR
+#
+
+####################################################################################
+# NOTE: This message bundle has only reserved CWWKL0090-CWWKL0099 if more messages #
+#       are needed, reserve a new block of messages in the classloading bundle     #
+####################################################################################
+cannot.apply.classloader.context=CWWKL0090E: Unable to apply classloader context to an unmanaged thread. Ensure that work is being submitted to a thread pool that is managed by the server.
```
Comment (njr-11) [inline dev/com.ibm.ws.classloader.context/resources/com/ibm/ws/classloader/context/internal/resources/ClassLoaderContextMessages.nlsprops]: 'submitted to a thread pool' is too specific.  There are also APIs for running with context on any existing thread, which could lead to this same problem.  'Ensure that work is being submitted to a thread pool that is managed by the server' is too limited, given the expected use cases for ContextService/ThreadContext.  Let's be more specific and say "Unable to apply class loader context to unmanaged thread {0}." specifically calling out the ForkJoinPool thread that blocks setContextClassLoader.  And we can expand the second sentence by switching to: "Ensure that work runs on a thread that is managed by the server or allows the setContextClassLoader operation."

File: dev/com.ibm.ws.classloader.context/resources/com/ibm/ws/classloader/context/internal/resources/ClassLoaderContextMessages.nlsprops [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+cannot.apply.classloader.context.explanation=Work was submitted to an unmanaged thread that does not support setting the ClassLoader
```
Comment (njr-11) [inline dev/com.ibm.ws.classloader.context/resources/com/ibm/ws/classloader/context/internal/resources/ClassLoaderContextMessages.nlsprops]: This sentence is good but is missing a . at the end.

File: dev/com.ibm.ws.classloader.context/resources/com/ibm/ws/classloader/context/internal/resources/ClassLoaderContextMessages.nlsprops [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+cannot.apply.classloader.context.useraction=Ensure that work is being submitted to a thread pool that is managed by the server.
```
Comment (njr-11) [inline dev/com.ibm.ws.classloader.context/resources/com/ibm/ws/classloader/context/internal/resources/ClassLoaderContextMessages.nlsprops]: Update per first comment

File: dev/com.ibm.ws.classloader.context/src/com/ibm/ws/classloader/context/internal/ClassloaderContextImpl.java [hunk 4]
```diff
@@ -144,14 +148,27 @@ public void taskStopping() {
...
-                Thread.currentThread().setContextClassLoader(cl);
+                final Thread t = Thread.currentThread();
+                try {
+                    t.setContextClassLoader(cl);
+                } catch (SecurityException e) {
+                    // If this work happens to run on an java.util.concurrent.ForkJoinWorkerThread$InnocuousForkJoinWorkerThread,
+                    // setting the ClassLoader may be rejected. If this happens, give a decent error message.
+                    if (t instanceof ForkJoinWorkerThread && "InnocuousForkJoinWorkerThreadGroup".equals(t.getThreadGroup().getName())) {
+                        String msg = Tr.formatMessage(tc, "cannot.apply.classloader.context");
+                        Tr.error(tc, msg);
```
Comment (njr-11) [inline dev/com.ibm.ws.classloader.context/src/com/ibm/ws/classloader/context/internal/ClassloaderContextImpl.java]: Does this show up properly in logs?  I suspect Tr.error will complain that it does not find a key matching the translated message.  That said, I think we should remove the Tr.error for this.  The exception already flows back to the application and is covered by FFDC.

Comment (aguibert) [reply dev/com.ibm.ws.classloader.context/src/com/ibm/ws/classloader/context/internal/ClassloaderContextImpl.java]: yep, it does.  I will remove the `Tr.error` though

**Comment by aguibert** [comment]
#build

**Comment by aguibert** [comment]
link to new build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_uWHcUOneEeiB5J8_tOvcPQ

**Comment by aguibert** [comment]
build is good and can be merged once dependencies used in build are promoted to master

**Pull request merged**